tests/FakeItEasy.Specs/CallMatchingSpecs.cs

```diff
@@ -631,22 +634,86 @@ public static void UnusedArgumentMatcherDescriptionNotUsed(IHaveNoGenericParamet
 
         [Scenario]
         public static void PassingIgnoredConstraintWithWrongTypeToAMethod(
-            IHaveNoGenericParameters fake, Exception exception, bool wasCalled)
+            IHaveNoGenericParameters fake,
+            Exception exception)
         {
             "Given a fake"
                 .x(() => fake = A.Fake<IHaveNoGenericParameters>());
 
             "When I try to configure a method of the fake with an Ignored constraint of the wrong type"
-                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(A<byte>.Ignored)).Invokes(() => wasCalled = true)));
+                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(A<byte>.Ignored))));
+
+            "Then the call configuration throws a FakeConfigurationException"
+                .x(() => exception.Should().BeAnExceptionOfType<FakeConfigurationException>());
+
+            "And the message should indicate the actual and expected type"
+                .x(() => exception.Message.Should().Be("Argument constraint is of type System.Byte, but parameter is of type System.Int32. No call can match this constraint."));
+        }
+
+        [Scenario]
+        public static void PassingHiddenConstraintWithWrongTypeToAMethod(
+            IHaveNoGenericParameters fake,
+            Func<int> constraintFactory,
+            Exception exception)
+        {
+            "Given a fake"
+                .x(() => fake = A.Fake<IHaveNoGenericParameters>());
+
+            "And a delegate that produces a constraint of the wrong type"
+                .x(() => constraintFactory = () => A<byte>.Ignored);
+
+            "When I try to configure a method of the fake with this delegate"
+                .x(() => exception = Record.Exception(() => A.CallTo(() => fake.Bar(constraintFactory()))));
+
+            "Then the call configuration throws a FakeConfigurationException"
+                .x(() => exception.Should().BeAnExceptionOfType<FakeConfigurationException>());
+
+            "And the message should indicate the actual and expected type"

```

Comment:
nit: previous steps used the imperative mood, not the subjunctive (fun fact: I would probably not even know these moods exist were it not for learning about "le subjontif" in school).
Consider rewording. Maybe something like "And the message indicates the actual and expected types"
(oh, I changed "type" to "types" there as well)

Reply:
Good point